test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,226 @@
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to update ingress controller: %v", err)
+	}
+
+	// Cleanup the ic.Spec.CompressionPolicy
+	defer func() {
+		ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+		if err != nil {
+			t.Fatalf("failed to cleanup ingress controller: %v", err)
+		}
+		mimeType := []operatorv1.CompressionMIMEType{}
+		ic.Spec.HTTPCompression = operatorv1.HTTPCompressionPolicy{MimeTypes: mimeType}
+		// Remove the mimeType that was added
+		if err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
+			if err := kclient.Update(context.TODO(), ic); err != nil {
+				t.Logf("failed to cleanup ingress controller: %v, retrying...", err)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("failed to cleanup ingress controller: %v", err)
+		}
+	}()
+
+	// Test some other compression policies for the ingress config
+	mimeTypesNormative := []operatorv1.CompressionMIMEType{"text/html", "application/json", "x-custom/allow-custom"}
+	compressionPolicyNormative := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNormative}
+
+	mimeTypesEmpty := []operatorv1.CompressionMIMEType{}
+	compressionPolicyEmpty := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesEmpty}
+
+	mimeTypesNeedQuotes := []operatorv1.CompressionMIMEType{`text/html; v="keepquoted"`, `x-custom/allow-custom; specialChar='`}
+	compressionPolicyNeedQuotes := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNeedQuotes}
+
+	mimeTypesErrors := []operatorv1.CompressionMIMEType{"text/", "x- /value", "//"}
+	compressionPolicyErrors := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesErrors}
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "http-compression-1"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	if err := testCompressionPolicy(t, icName, domain, compressionPolicyNormative); err != nil {
+		log.Fatalf("error testing normal compression policy: %v", err)
+	}
+
+	icName = types.NamespacedName{Namespace: operatorNamespace, Name: "http-compression-2"}
+	domain = icName.Name + "." + dnsConfig.Spec.BaseDomain
+	if err := testCompressionPolicy(t, icName, domain, compressionPolicyEmpty); err != nil {
+		log.Fatalf("error testing empty compression policy: %v", err)
+	}
+
+	icName = types.NamespacedName{Namespace: operatorNamespace, Name: "http-compression-3"}
+	domain = icName.Name + "." + dnsConfig.Spec.BaseDomain
+	if err := testCompressionPolicy(t, icName, domain, compressionPolicyNeedQuotes); err != nil {
+		log.Fatalf("error testing compression policy that needs quotes: %v", err)
+	}
+
+	icName = types.NamespacedName{Namespace: operatorNamespace, Name: "http-compression-4"}
+	domain = icName.Name + "." + dnsConfig.Spec.BaseDomain
+	if err := testCompressionPolicy(t, icName, domain, compressionPolicyErrors); err == nil {
+		log.Fatalf("compression policy with errors should have failed but didn't")
+	}
```

Comment:
It seems like this block is one test, and the lines above this block and the lines below this block are another test.  Could this block be moved to the end of the test function or to a separate test function?

Reply:
yes, thanks.